enumerables.rb

```diff
@@ -0,0 +1,139 @@
+module Enumerable
+  # 1. Create my_each
+  def my_each
+    (0..length - 1).each do |i|
+      yield self[i]
+    end
+  end
+
+  # 2. Create my_each_with_index
+  def my_each_with_index
+    (0..length - 1).each do |i|
+      yield(self[i], i)
+    end
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`each_with_index`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each_with_index.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] returns an enumerator if no block is given
```ruby 
      my_each_output = ''
      block = proc { |num, idx| my_each_output += "Num: #{num}, idx: #{idx}\n" }
      array.each_with_index(&block)
      each_output = my_each_output.dup
      my_each_output = ''
      array.my_each_with_index(&block)
      my_each_output == each_output #true
```